doc/lspconfig.txt

```diff
@@ -1,242 +1,628 @@
+      buf_set_keymap('n', '[d', '<cmd>lua vim.lsp.diagnostic.goto_prev()<CR>', opts)
+      buf_set_keymap('n', ']d', '<cmd>lua vim.lsp.diagnostic.goto_next()<CR>', opts)
+      buf_set_keymap('n', '<space>q', '<cmd>lua vim.lsp.diagnostic.set_loclist()<CR>', opts)
+      buf_set_keymap('n', '<space>f', '<cmd>lua vim.lsp.buf.formatting()<CR>', opts)
+
+    end
+
+    -- Use a loop to conveniently call 'setup' on multiple servers and
+    -- map buffer local keybindings when the language server attaches
+    local servers = { 'pyright', 'rust_analyzer', 'tsserver' }
+    for _, lsp in ipairs(servers) do
+      nvim_lsp[lsp].setup {
+        on_attach = on_attach,
+        flags = {
+          debounce_text_changes = 150,
+        }
+      }
+    end
+<
+Note: these keymappings are illustrative and override some infrequently used
+default mappings.
+
+==============================================================================
+COMPLETION SUPPORT                                        *lspconfig-completion*
+
+Manual, triggered completion can be provided by neovim's built-in omnifunc.
+See `:help omnifunc` for more details.
+
+For autocompletion, neovim does not offer built-in functionality at this time.
+Consult the lspconfig wiki, which provides configuration examples for using a
+completion plugin with the built-in client
+
+==============================================================================
+DEBUGGING                                                  *lspconfig-debugging*
+
+While using language servers should be easy, debugging issues can be
+challenging. It is first important to identify the source of the issue, which
+is typically (in rough order):
+
+- the language server itself
+- a plugin
+- overrides in a user configuration
+- the built-in client in neovim core
+- lspconfig
+
+The first step in debugging is to test with a minimal configuration (such as
+`../test/minimal_init.lua`). Historically, many users problems are due to
+plugins or misconfiguration.
+
+Should that fail, identifying which component is the culprit is challenging.
+The following are the *only* categories of bugs that pertain to lspconfig.
+
+- The root directory inferred for your project is wrong. Or it should be
+  detected but is not based on a flaw in the lspconfig path utilities.
+- The server is launching, but you believe that the default settings,
+  initialization options, or command arguments are suboptimal and should be
+  replaced based on your understanding of the server documentation.
+
+All bugs pertaining to errors in core neovim client should be reported to the
+neovim core issue tracker. All bugs pertaining to plugins should be reported
+to the respective plugin. All missing features in a language server should be
+reported to the upstream language server issue tracker.
+
+For debugging lspconfig issues the most common mistakes users make are:
```

Comment:
* comma before "the"

* maybe less accusatory: "the most common stumbling blocks" / "hurdles"